plugin/storage/grpc/shared/grpc_server.go

```diff
@@ -53,7 +64,8 @@ func (s *grpcServer) WriteSpan(ctx context.Context, r *storage_v1.WriteSpanReque
 
 // GetTrace takes a traceID and streams a Trace associated with that traceID
 func (s *grpcServer) GetTrace(r *storage_v1.GetTraceRequest, stream storage_v1.SpanReaderPlugin_GetTraceServer) error {
-	trace, err := s.Impl.SpanReader().GetTrace(stream.Context(), r.TraceID)
+	opCtx := s.getMaybeBearerTokenContext(stream.Context(), r.BearerToken)
```

Comment:
this may not be needed per my earlier comments, but if needed then it would be better to rename to a more general method of "adapting" the context from request, so that we could, in the future, support different types of middleware.